Vizzuality/heco-invest
# feat: Finalisation of klab poll_impact_demands job

Finalisation of `Klab::PollImpactDemandsJob` so it:
 - poll status of ticket from Klab
 - export artifact and saves demands to project table if ticket is already resolved
 - waits if ticket is still open --> by default there is `250` attempts with 10s of waiting time per try (overall waiting time is around 45 minutes)

I have also refactored a bit `backend/app/services/klab/poll_ticket.rb` so you can choose if polling is gonna be async (waiting is handled by job) or not (waits in cycle till ticket gets resolved). I have decided to use this approach because I wanted to keep `CalculateProjectImpactScore` working.

## Testing instructions

You can use `Klab::PollImpactDemandsJob` to obtain demands from BC3 which gets saved at project table. To manually test this, you will first need to submit context to BC3 which gives you appropriate `ticket_id` which is mandatory param when initialising this job. There are multiple possible scenarios:
 - ticket got resolved --> saves demands at project table
 - ticket is still open --> enqueues new polling job and waits
 - polling job run out of attempts (waited for too long) --> error gets raised
 
All use cases are covered by unit tests, but it should be also possible to test them manually...

## Tracking

https://vizzuality.atlassian.net/browse/LET-1354
